docker/pypi/wmagent/bin/manage-common.sh

```diff
@@ -0,0 +1,423 @@
+
+    # Here to find out if the agent is CERN or FNAL and use the proper credentials name for _renew_proxy
+    if [[ "$hostName" == *cern.ch ]]; then
+        local myproxyCredName="amaltaroCERN"
+    elif [[ "$hostName" == *fnal.gov ]]; then
+        local myproxyCredName="amaltaroFNAL"
+    else
+        echo "$FUNCNAME: ERROR: Sorry, we do not recognize the network domain of the current host: $hostName"
+        return $(false)
+    fi
+
+    # Here to forge the myproxy command string to be used for the operation.
+    local myproxyCmd="myproxy-get-delegation \
+                    -v -l amaltaro -t 168 -s myproxy.cern.ch -k $myproxyCredName -n \
+                    -o $WMA_CERTS_DIR/mynewproxy.pem"
+    local vomsproxyCmd="voms-proxy-init -rfc \
+                    -voms cms:/cms/Role=production -valid 168:00 -bits 2048 -noregen \
+                    -cert $WMA_CERTS_DIR/mynewproxy.pem \
+                    -key  $WMA_CERTS_DIR/mynewproxy.pem \
+                    -out  $WMA_CERTS_DIR/myproxy.pem"
+
+    # Here to check certificates and proxy lifetime and update myproxy if needed:
+    local certMinLifetimeHours=168
+    local certMinLifetimeSec=$(($certMinLifetimeHours*60*60))
+
+    if [[ -f $WMA_CERTS_DIR/servicecert.pem ]] && [[ -f $WMA_CERTS_DIR/servicekey.pem ]]; then
+
+        echo "$FUNCNAME: Checking Certificate lifetime:"
+        local now=$(date +%s)
+        local certEndDate=$(openssl x509 -in $WMA_CERTS_DIR/servicecert.pem -noout -enddate)
+        certEndDate=${certEndDate##*=}
+        echo "$FUNCNAME: Certificate end date: $certEndDate"
+        [[ -z $certEndDate ]] && {
+            echo "$FUNCNAME: ERROR: Failed to determine certificate end date!"; return $(false) ;}
+        certEndDate=$(date --date="$certEndDate" +%s)
+        [[ $certEndDate -le $now ]] && {
+            echo "$FUNCNAME: ERROR: Expired certificate at $WMA_CERTS_DIR/servicecert.pem!"; return $(false) ;}
+        [[ $(($certEndDate -$now)) -le $certMinLifetimeSec ]] && {
+            echo "$FUNCNAME: WARNING: The service certificate lifetime is less than certMinLifetimeHours: $certMinLifetimeHours! Please update it ASAP!" ;}
+
+        # Renew myproxy if needed:
+        echo "$FUNCNAME: Checking myproxy lifetime:"
+        local myproxyEndDate=$(openssl x509 -in $WMA_CERTS_DIR/myproxy.pem -noout -enddate)
+        myproxyEndDate=${myproxyEndDate##*=}
+        echo "$FUNCNAME: myproxy end date: $myproxyEndDate"
+        [[ -n $myproxyEndDate ]] || $myproxyCmd && $vomsproxyCmd || {
+                echo "$FUNCNAME: ERROR: Failed to renew invalid myproxy"; return $(false) ;}
+        myproxyEndDate=$(date --date="$myproxyEndDate" +%s)
+        [[ $myproxyEndDate -gt $now ]] || $myproxyCmd && $vomsproxyCmd || {
+                echo "$FUNCNAME: ERROR: Failed to renew expired myproxy"; return $(false) ;}
+
+        # Stay safe and always change the service {cert,key} and myproxy mode here:
+        sudo chmod 600 $WMA_CERTS_DIR/*
```

Comment:
We need to check, but it might be that rucio-clients - or one of its dependencies - will not work if the key.pem file has this permission (contrary to 400).

Reply:
Rucio clients  do work at the moment, but actually `400` are the better file permissions here, which are the recommended  ones in general. Fixing it.

Reply:
Thanks for bringing that @amaltaro. I actually found  a bug while testing this file permissions - I have forgotten to reorder the logical conditions chain such that the exit value from the innermost `&&` to be conciderred as a whole, because we want both commands `$myproxyCmd` and `vomsproxyCmd` to succeed in this operation. Fixing that one as well.

Reply:
AFAIK public key needs to be 0x600, while the private one 0x400. You might want to test things out before changing it.

Reply:
I did run the agent with the new changes. So far things  seem to be working just fine.